### PR TITLE
Add register uniqueness test

### PR DESCRIPTION
**Pull request by hainest**
This ensures that all register values are unique for a given architecture. ppc32 is explicitly not tested because they are not supported.

### CODE REVIEW

**Comment by hainest** [comment]
This test will fail until I get the arm register update merged, so let's hold off on merging this until then.

**Comment by hainest** [comment]
@kupsch if/when you have time.

**Review by kupsch — APPROVED** [review]
LGTM.

**Pull request merged**
